### ISSUE TITLE
Feature Request: DefineConstants

### ISSUE THREAD

**Issue by jnyrup** [stryker-mutator/stryker-net#522]
**Is your feature request related to a problem? Please describe.**
I'm using preprocessor directives to do different implementations when multi-targeting libraries.

Example:
`Library.csproj`
```xml
<PropertyGroup Condition="'$(TargetFramework)' == 'net472'">
    <DefineConstants>NET472</DefineConstants>
</PropertyGroup>
```
`File.cs`
```cs
#if NET472
    public void Method() { }
#endif
}
```

Currently when using Stryker 0.11.0 the log contains the following error when testing a mutant:
```
Starting test execution, please wait...
Failed   TestMethod1
Error Message:
 Test method StryknerTests.UnitTest1.TestMethod1 threw exception: 
System.MissingMethodException: Method not found: 'Int32 StryknerConstants.Class1.Add(Int32, Int32)'.
Stack Trace:
    at StryknerTests.UnitTest1.TestMethod1()
```

**Describe the solution you'd like**
The mutants should be compiled with the same constants as when compiling the project manually.

**Describe alternatives you've considered**
For the project I tried to analyze my workaround was to replace `NET472` in preprocessor directives with `(NET472 || true)`.

**Additional context**
Full log:[log-20190520.txt](https://github.com/stryker-mutator/stryker-net/files/3199207/log-20190520.txt)
Full example: [StrykerConstants.zip](https://github.com/stryker-mutator/stryker-net/files/3199211/StrykerConstants.zip)

**Comment by Mobrockers** [comment]
This request is similar to #466 
The proposed solution then was to let the user define the msbuild argument in config manually but that only works for the initial build so we'll have to figure out how this would work when compiling directly with Roslyn.

**Comment by Mobrockers** [comment]
@richardwerkman  We should investigate whether buildalyzer is capable of giving us all preprocessor directives in a way we can easily use without having to build out every single case.

**Comment by Mobrockers** [comment]
@richardwerkman We can borrow code from buildalyzer to implement this:

https://github.com/daveaglick/Buildalyzer/commit/df80c42ec909eded91f5b6839a6f632beff2aa2a

**Comment by Mobrockers** [comment]
This was actually fixed in #789 without noticing! @jnyrup Could you test if the current defineconstants implementation fixes the issues you had?

**Comment by jnyrup** [comment]
I can confirm that this was fixed between 0.14.0 (didn't work) and 0.14.2 (worked).
Thanks!

**Comment by dupdob** [comment]
This was expected, I had to add support for symbols for 789.  I was not aware there was an open issue for this. 
🤷

Envoyé de mon iPhone
